dotCMS/src/main/java/com/dotcms/util/pagination/FolderContentReportPaginator.java

```diff
@@ -0,0 +1,78 @@
+package com.dotcms.util.pagination;
+
+import com.dotcms.exception.ExceptionUtil;
+import com.dotcms.rest.api.v1.content.ContentReportHelper;
+import com.dotcms.rest.api.v1.content.ContentReportParams;
+import com.dotcms.rest.api.v1.content.ContentReportView;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PaginatedArrayList;
+import com.liferay.portal.model.User;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This implementation of the {@link ContentReportPaginator} provides the Content Report for all the
+ * Content Types living under a specific Folder in dotCMS. The generated report analyzes all the
+ * child Folders inside the specified one as well.
+ *
+ * @author Jose Castro
+ * @since Mar 8th, 2024
+ */
+public class FolderContentReportPaginator extends ContentReportPaginator<ContentReportView> {
+
+    private final ContentReportHelper contentReportHelper;
+
+    public FolderContentReportPaginator(final User user) {
+        this.contentReportHelper = new ContentReportHelper(user);
```

Comment:
after the comment above this would become: 
```this.contentReportHelper = new FolderContentReportHelper(user);```

Reply:
Done!